services/backend/src/assignment-controller.js

```diff
@@ -0,0 +1,107 @@
+const _ = require('lodash');
+const { AWS } = require('@logan/aws');
+const { v4: uuid } = require('uuid');
+
+const dynamo = new AWS.DynamoDB.DocumentClient();
+
+function fromDbFormat(db) {
+    return {
+        ..._.pick(db, ['aid', 'uid', 'title', 'desc', 'due', 'cid']),
+        description: db.desc,
+        dueDate: db.due,
+    };
+}
+
+function toDbFormat(assignment) {
+    return {
+        ..._.pick(assignment, ['aid', 'uid', 'title', 'desc', 'due', 'cid']),
+        desc: assignment.description,
+        due: assignment.dueDate,
+    };
+}
+
+async function getAssignment(req, res) {
+    const requestedAid = req.params.aid;
+
+    const dbResponse = await dynamo
+        .get({
+            TableName: 'assignments',
+            Key: { aid: requestedAid },
+        })
+        .promise();
+
+    if (dbResponse.Item) {
+        res.json(fromDbFormat(dbResponse.Item));
+    } else {
+        throw new Error('Assignment does not exist');
+    }
+}
+
+async function getAssignments(req, res) {
+    const requestedBy = req.auth.uid;
+
+    const dbResponse = await dynamo
+        .scan({
+            TableName: 'assignments',
+            FilterExpression: 'uid = :uid',
+            ExpressionAttributeValues: {
+                ':uid': requestedBy,
+            },
+        })
+        .promise();
+
+    res.json(dbResponse.Items.map(fromDbFormat(dbResponse.Items)));
```

Comment:
```suggestion
    res.json(dbResponse.Items.map(fromDbFormat));
```

You're passing the mapping function as the parameter